### PR TITLE
Reference Assemblies for readonly structs

### PR DESCRIPTION
**Pull request by benaadams**
Follow up to https://github.com/dotnet/coreclr/pull/19552 and https://github.com/dotnet/coreclr/pull/19557

Can't see a reference assembly for `AsyncLocalValueChangedArgs` or `HandleRef`

/cc @jkotas

### CODE REVIEW

**Comment by jkotas** [comment]
> HandleRef

src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

> AsyncLocalValueChangedArgs 

src/System.Threading/ref/System.Threading.cs

**Comment by benaadams** [comment]
Ah, thanks added.

**Comment by benaadams** [comment]
```
Type 'System.DateTimeOffset' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
Type 'System.TimeSpan' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
Type 'System.Runtime.Serialization.SerializationEntry' is marked as readonly in the contract so it must also be marked readonly in the implementation.
```
But https://github.com/dotnet/corefx/pull/31844 and https://github.com/dotnet/corefx/pull/31843 are merged?

Needs coreclr update, rather than just code mirror?

**Comment by jkotas** [comment]
> Needs coreclr update, rather than just code mirror?

Right.

**Comment by danmosemsft** [comment]
Is this going to be wiped out next time we run `msbuild /t:GenerateReferenceSource` ? If so, we need to update genapis. And if not, presumably you can simply run that command.

**Comment by jkotas** [comment]
genapi was updated for readonly structs long time ago (https://github.com/dotnet/buildtools/pull/1849).

**Comment by danmosemsft** [comment]
> genapi was updated for readonly structs long time ago (dotnet/buildtools#1849).

I'm missing something then, as when I run it, it does not make the change @benaadams is making.

**Comment by jkotas** [comment]
We need to pick up updated CoreCLR first. The update PR https://github.com/dotnet/corefx/pull/31853 was stuck on buildtools breaks. I have rolled back the buildtools breaks to get it unstuck.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by benaadams** [comment]
UWP NETNative doesn't like it
```
Type 'System.DateTimeOffset' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
Type 'System.TimeSpan' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
Type 'System.Runtime.Serialization.SerializationEntry' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
 error : ApiCompat failed for 'D:\j\workspace\windows-TGrou---c60886e1\bin\Windows_NT.AnyCPU.Release\System.Runtime\uapaot\System.Runtime.dll' 
```

**Comment by jkotas** [comment]
It takes longer for the UWP NETNative changes to propagate. You add this to baseline files like
https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/ApiCompatBaseline.uapaot.txt to make the CI green.

File: src/System.Runtime/src/ApiCompatBaseline.uapaot.txt [hunk 1]
```diff
@@ -2,3 +2,6 @@ Compat issues with assembly System.Runtime:
 TypeCannotChangeClassification : Type 'System.Decimal' is marked as readonly in the contract so it must also be marked readonly in the implementation.
 TypeCannotChangeClassification : Type 'System.Double' is marked as readonly in the contract so it must also be marked readonly in the implementation.
 TypeCannotChangeClassification : Type 'System.Single' is marked as readonly in the contract so it must also be marked readonly in the implementation.
+TypeCannotChangeClassification : Type 'System.DateTimeOffset' is marked as readonly in the contract so it must also be marked readonly in the implementation.
+TypeCannotChangeClassification : Type 'System.TimeSpan' is marked as readonly in the contract so it must also be marked readonly in the implementation.
```
Comment (jkotas) [inline src/System.Runtime/src/ApiCompatBaseline.uapaot.txt]: src/System.Runtime.InteropServices/src/ApiCompatBaseline.uapaot.txt and src/System.Threading/src/ApiCompatBaseline.uapaot.txt will need similar change.

File: src/System.Runtime/src/ApiCompatBaseline.uapaot.txt [hunk 2]
```diff
@@ -2,3 +2,6 @@ Compat issues with assembly System.Runtime:
 TypeCannotChangeClassification : Type 'System.Decimal' is marked as readonly in the contract so it must also be marked readonly in the implementation.
```
Comment (jkotas) [inline src/System.Runtime/src/ApiCompatBaseline.uapaot.txt]: Nit: You can delete the existing content while you are on it. It should not be necessary anymore.

**Comment by benaadams** [comment]
Do I need to add something in for Packaging? Its reporting same errors
```
TypeCannotChangeClassification : Type 'System.DateTimeOffset' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
TypeCannotChangeClassification : Type 'System.TimeSpan' is marked as readonly in the contract so it must also be marked readonly in the implementation. 
TypeCannotChangeClassification : Type 'System.Runtime.Serialization.SerializationEntry' is marked as readonly in the contract so it must also be marked readonly in the implementation.
ApiCompat failed for 'D:\j\workspace\windows-TGrou---0d2c9ac4\bin\Windows_NT.AnyCPU.Debug\System.Runtime\netcoreappaot\System.Runtime.dll' 
```

**Comment by jkotas** [comment]
That's the netcoreappaot build flavor that @morganbr added yesterday. Just make the same changes in  ApiCompatBaseline.netcoreappaot.txt files. I wondering whether we can do something about these duplicated baselines.

**Comment by morganbr** [comment]
That sounds nice. @ericstj, might you have any ideas on consolidating baselines?

**Comment by morganbr** [comment]
It's the same codebase except for some ifdefs. The important differences are that netcoreappaot doesn't expose WinRT functionality and there are some differences in the uap and netcoreapp surface area that might be visible (for example, compiled regex is in the netcoreapp surface area).

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**